ompi/mpi/fortran/base/gen-mpi-mangling.pl

```diff
@@ -193,7 +226,15 @@ sub gen_f08_types {
 
     foreach my $key (sort(keys(%{$fortran}))) {
         my $f = $fortran->{$key};
-        print OUT "$f->{f_type}, bind(C, name=\"".mangle($f->{c_name})."\") :: $f->{f_name}\n";
+        my $m = mangle($f->{c_name});
+        my @all_names = mangle_all($f->{c_name});
+# I think we can only bind each var to one place, so I'm short-circuiting the
+# above to replace @all_names with its first element.
+        @all_names = ($all_names[0]);
+        foreach my $n (@all_names) {
+            print OUT "$f->{f_type}, bind(C, name=\"".$n."\") :: $f->{f_name}\n";
+        }
+        print OUT "\n";
```

Comment:
I don't understand this perl block.  In the description for this PR, you said:

> It had been generating
>  integer, bind(C, name="mpi_fortran_in_place") :: MPI_IN_PLACE
>  integer, bind(C, name="MPI_FORTRAN_IN_PLACE") :: MPI_IN_PLACE
>  integer, bind(C, name="mpi_fortran_in_place_") :: MPI_IN_PLACE
>  integer, bind(C, name="mpi_fortran_in_place__") :: MPI_IN_PLACE
> but I think it's only possible to bind a variable to one place.
> And for F08 I think that's okay that we can only make one binding.

But I don't see that happening in my build of Open MPI with the `mpi_f08` module.  My `mpif-f08-types.h` (on master) is:

``` fortran
character, dimension(1), bind(C, name="mpi_fortran_argv_null_") :: MPI_ARGV_NULL
character, dimension(1, 1), bind(C, name="mpi_fortran_argvs_null_") :: MPI_ARGVS_NULL
integer, bind(C, name="mpi_fortran_bottom_") :: MPI_BOTTOM
integer, dimension(1), bind(C, name="mpi_fortran_errcodes_ignore_") :: MPI_ERRCODES_IGNORE
integer, bind(C, name="mpi_fortran_in_place_") :: MPI_IN_PLACE
type(MPI_STATUS), bind(C, name="mpi_fortran_status_ignore_") :: MPI_STATUS_IGNORE
type(MPI_STATUS), bind(C, name="mpi_fortran_statuses_ignore_") :: MPI_STATUSES_IGNORE(1)
integer, dimension(1), bind(C, name="mpi_fortran_unweighted_") :: MPI_UNWEIGHTED
integer, dimension(1), bind(C, name="mpi_fortran_weights_empty_") :: MPI_WEIGHTS_EMPTY
```

If you're getting 4 bindings to `MPI_IN_PLACE`, is that because of how Spectrum MPI does some magic to generate Fortran symbols for all 4 old-style linker symbols?